tests/e2e/default/service-worker/service-worker.wdio-spec.js

```diff
@@ -78,15 +78,30 @@ const login = async () => {
   await commonPage.waitForPageLoaded();
 };
 
+const isLoggedIn = async () => {
+  const tab = await commonPage.messagesTab();
+  return await tab.isExisting();
+};
+
+const loginIfNeeded = async () => {
+  if (!await isLoggedIn()) {
+    await login();
+  }
+};
+
 describe('Service worker cache', () => {
+
   before(async () => {
     await utils.saveDoc(district);
     await utils.createUsers([chw]);
-
     await login();
     await commonPage.closeTour();
   });
 
+  beforeEach(async () => {
+    await loginIfNeeded();
```

Comment:
Having the login here means it re-runs on retry so can pass the second time around.